### ISSUE TITLE
AccessViolation trying to compile Method

### ISSUE THREAD

**Issue by Suchiman** [dotnet/runtime#31615]
```csproj
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp3.1</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="SixLabors.ImageSharp" Version="1.0.0-beta0006" />
  </ItemGroup>
  
</Project>
```
```csharp
class Program
{
    static void Main(string[] args)
    {
        SixLabors.ImageSharp.PixelFormats.ColorBuilder<SixLabors.ImageSharp.PixelFormats.RgbaVector>.FromHex("test");
    }
}
```

Repro steps:
1. Publish the above as standalone (`dotnet publish -c Release -r win-x64`)
2. Overwrite the runtime and framework in the standalone publish with the current master build from dotnet/runtime
3. `set COMPlus_TieredCompilation=0`
4. Run the program

Result:
```
Fatal error. System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
   at SixLabors.ImageSharp.PixelFormats.ColorBuilder`1[[SixLabors.ImageSharp.PixelFormats.RgbaVector, SixLabors.ImageSharp, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].FromHex(System.String)
   at Program.Main(System.String[])
```

**Comment by jkotas** [comment]
The same crash reproduces on https://github.com/dotnet/runtime

**Comment by jkotas** [comment]
cc @dotnet/jit-contrib 

This is a recent regression. It works fine in .NET Core 3.1. It may be a duplicate of #13643 (hitting same assert).

@Suchiman Thank you for reporting this!

**Comment by AndyAyersMS** [comment]
Took a quick look; morph coalesces a series of copies into a single SIMD copy, later the merged-up copy dest confuses value numbering:
```
;; morph
Found contiguous assignments from a SIMD vector to memory.
From BB57, stmtSTMT00077 to stmtSTMT00080

BB57 stmtSTMT00077(before)
STMT00077 (IL 0x04F...  ???)
               [000382] -A----------              *  ASG       float  (copy)
               [000381] U------N----              +--*  LCL_FLD   simd16 V03 loc1         [+0]
               [000380] ------------              \--*  FIELD     float  X
               [000379] ------------                 \--*  ADDR      byref 
               [000378] ------------                    \--*  LCL_VAR   simd16<System.Numerics.Vector4> V27 tmp22        
LocalAddressVisitor visiting statement:
STMT00077 (IL 0x04F...  ???)
               [000851] -A----------              *  ASG       simd16 (copy)
               [000381] U------N----              +--*  LCL_FLD   simd16 V03 loc1         [+0]
               [000378] ------------              \--*  LCL_VAR   simd16<System.Numerics.Vector4> V27 tmp22        
...
;; valuenum

***** BB24, STMT00077(before)
N003 (  5,  6) [000851] -A--G---R---              *  ASG       simd16 (copy)
N002 (  3,  4) [000381] U------N----              +--*  LCL_FLD   simd16 V03 loc1         ud:3->0[+0]
N001 (  1,  1) [000378] ------------              \--*  LCL_VAR   simd16<System.Numerics.Vector4> V43 tmp38        u:2 (last use)

N001 [000378]   LCL_VAR   V43 tmp38        u:2 (last use) => $750 {750}
N002 [000381]   LCL_FLD   V03 loc1         ud:3->0[+0] => $751 {751}
    *** Missing field sequence info for Dst/LHS of COPYBLK
;; Fails looking for SSA def of [000381] 
```

**Comment by mikedn** [comment]
Looks like fallout from my #737. I'll work on a fix.

**Comment by mikedn** [comment]
I have a fix in #31626 but the newly added test fails on ARM64 with:
```
Assert failure(PID 2767 [0x00000acf], Thread: 2767 [0x0acf]): Assertion failed 'argx->OperIs(GT_LCL_VAR)' in 'RgbaVector:FromVector4(System.Numerics.Vector4):this' (IL size 67)

    File: /__w/2/s/src/coreclr/src/jit/morph.cpp Line: 4190
    Image: /root/helix/work/correlation/corerun
```
https://dev.azure.com/dnceng/public/_build/results?buildId=504921&view=ms.vss-test-web.build-test-results-tab&runId=16031954&resultId=100805&paneView=attachments

I'm looking into it, so far it looks like an unrelated bug.

**Comment by mikedn** [comment]
Filled #31673 for the above assert and modified the new test to workaround it. It's a preexisting issue (from 3.x at least), unrelated to my changes, so I'll leave it to the team to fix it.
